### PR TITLE
add commit file statuses

### LINKED ISSUE #4313: Webhook POST data doesn't pass added/modified/removed info in commits section
- Gitea version (or commit ref): 1.4.2
- Git version: 2.13
- Operating system: Linux
- Database (use `[x]`): 
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gitea.io: (URL is down at the time of issue)
  - [X] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
I created a web hook and pushed a change (Tried with both content types json and xml)
When a looked at payload sent by gitea in commit section 
"commits": [
    {
      "id": "bffeb74224043ba2feb48d137756c8a9331c449a",
      "message": "Webhooks Yay!",
      "url": "http://localhost:3000/gitea/webhooks/commit/bffeb74224043ba2feb48d137756c8a9331c449a",
      "author": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "committer": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "timestamp": "2017-03-13T13:52:11-04:00"
    }
  ]

there is no added/modified/removed info

"commits": [
    {
      "id": "bffeb74224043ba2feb48d137756c8a9331c449a",
      "message": "Webhooks Yay!",
      "url": "http://localhost:3000/gitea/webhooks/commit/bffeb74224043ba2feb48d137756c8a9331c449a",
      "author": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "committer": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "timestamp": "2017-03-13T13:52:11-04:00"
    }
    "added": ...

  ]

### PR DESCRIPTION
**Pull request by lunny**
Should fix #4313 and ~blocked by https://github.com/go-gitea/go-sdk/pull/135 and https://github.com/go-gitea/git/pull/137~

### CODE REVIEW

**Comment by lafriks** [comment]
Dependencies has been merged

**Comment by lunny** [comment]
@lafriks updated

**Comment by lafriks** [comment]
Do not add newline in the end of `v1_json.tmpl` otherwise drone will fail

**Comment by lunny** [comment]
@lafriks done.

**Comment by decke** [comment]
Is there any work needed before it can get merged?

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
    the top of the list. Name must match file name in conf/license or custom/conf/license.
 - `DISABLE_HTTP_GIT`: **false**: Disable the ability to interact with repositories over the
    HTTP protocol.
+- `ACCESS_CONTROL_ALLOW_ORIGIN`: ****: Value for Access-Control-Allow-Origin header, default is not to present.
```
Comment (techknowlogick) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]: I think this is for a different PR?

Comment (zeripath) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
```

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
...
+   WARNING: This maybe harmful to you website if you do not give it a right value.
```
Comment (zeripath) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
```

File: docs/content/doc/advanced/config-cheat-sheet.zh-cn.md [hunk 3]
```diff
@@ -32,6 +32,8 @@ menu:
 - `DEFAULT_PRIVATE`: 默认创建的git工程为私有。 可以是`last`, `private` 或 `public`。默认值是 `last`表示用户最后创建的Repo的选择。
 - `MAX_CREATION_LIMIT`: 全局最大每个用户创建的git工程数目， `-1` 表示没限制。
 - `PULL_REQUEST_QUEUE_LENGTH`: 小心：合并请求测试队列的长度，尽量放大。
+- `ACCESS_CONTROL_ALLOW_ORIGIN`: ****: Access-Control-Allow-Origin 头的值，默认为空。
```
Comment (zeripath) [inline docs/content/doc/advanced/config-cheat-sheet.zh-cn.md]:
```suggestion
```

File: docs/content/doc/advanced/config-cheat-sheet.zh-cn.md [hunk 4]
```diff
@@ -32,6 +32,8 @@ menu:
...
+   警告：如果您不熟悉这方面的设置请不要修改此值，可能影响网站安全。
```
Comment (zeripath) [inline docs/content/doc/advanced/config-cheat-sheet.zh-cn.md]:
```suggestion
```

**Pull request closed without merging**